dd-java-agent/agent-debugger/debugger-el/src/main/java/com/datadog/debugger/el/values/SetValue.java

```
@@ -0,0 +1,86 @@
+package com.datadog.debugger.el.values;
+
+import com.datadog.debugger.el.Value;
+import com.datadog.debugger.el.Visitor;
+import com.datadog.debugger.el.expressions.ValueExpression;
+import datadog.trace.bootstrap.debugger.el.ValueReferenceResolver;
+import datadog.trace.bootstrap.debugger.el.Values;
+import datadog.trace.bootstrap.debugger.util.WellKnownClasses;
+import java.util.Set;
+
+public class SetValue implements CollectionValue<Object>, ValueExpression<SetValue> {
+
+  private final Object setHolder;
+
+  public SetValue(Object object) {
+    if (object instanceof Set) {
+      setHolder = object;
+    } else if (object == null || object == Values.NULL_OBJECT) {
+      setHolder = Value.nullValue();
+    } else {
+      setHolder = Value.undefinedValue();
+    }
+  }
+
+  @Override
+  public SetValue evaluate(ValueReferenceResolver valueRefResolver) {
+    return this;
+  }
+
+  @Override
+  public Object getValue() {
+    return setHolder;
+  }
+
+  @Override
+  public boolean isEmpty() {
+    if (setHolder instanceof Set) {
+      return ((Set<?>) setHolder).isEmpty();
+    } else if (setHolder instanceof Value) {
+      Value<?> val = (Value<?>) setHolder;
+      return val.isNull() || val.isUndefined();
+    }
+    return true;
+  }
+
+  @Override
+  public int count() {
+    if (setHolder instanceof Set) {
+      if (WellKnownClasses.isSizeSafe((Set<?>) setHolder)) {
+        return ((Set<?>) setHolder).size();
+      }
+      throw new RuntimeException("Unsupported Set class: " + setHolder.getClass().getTypeName());
+    } else if (setHolder == Value.nullValue()) {
+      return 0;
+    }
+    return -1;
```

why -1? why not undefined? is that in the spec ?
I think count on String also return -1 for null. not sure why we do that.

---

legacy from the initial implementation. should never reached this code.
> why not undefined?
because method should return an int! otherwise we need to throw an exception